test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -315,14 +300,16 @@
            (is (true? (set-constants-and-verify pool-context 3)))
 
            ;; The flush should still not have completed at this point because
-           ;; the last instance from the old pool is still being borrowed
-           (is (not (realized? flush-complete)))
+           ;; the last instance from the old pool is still being borrowed.
+           ;; Wait for some arbitrary (but short) period of time to see that
+           ;; that the agent is still busy - presumably handling the flush.
+           (is (false? (await-for 50 pool-agent)))
```

Comment:
I agree there's a risk this assertion may pass even if the flush completes sometime after this line is run and the subsequent `return-instance` call is made.  If you have an idea on how to make that better, I'm definitely open to doing that.